src/main/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/network/v5_4/OpenNebulaNetworkPlugin.java

```diff
@@ -79,50 +81,26 @@ public boolean hasFailed(String cloudState) {
 	public String requestInstance(NetworkOrder networkOrder, CloudUser cloudUser) throws FogbowException {
 		Client client = OpenNebulaClientUtil.createClient(this.endpoint, cloudUser.getToken());
 		VirtualNetwork virtualNetwork = OpenNebulaClientUtil.getVirtualNetwork(client, this.defaultNetwork);
+		String cidr = networkOrder.getCidr();
+		SubnetUtils.SubnetInfo subnetInfo = new SubnetUtils(cidr).getInfo();
 
-		int defaultNetworkID = convertToInteger(this.defaultNetwork);
 		String name = networkOrder.getName();
-
-		SubnetUtils.SubnetInfo subnetInfo = new SubnetUtils(networkOrder.getCidr()).getInfo();
 		int size = subnetInfo.getAddressCount();
-		Integer addressRangeIndex = this.getAddressRangeIndex(virtualNetwork, subnetInfo.getLowAddress(), size);
-		String addressRangeId = null;
-
-		if (addressRangeIndex != null) {
-			addressRangeId = virtualNetwork.xpath(String.format(ADDRESS_RANGE_ID_PATH_FORMAT, addressRangeIndex));
-		} else {
-			throw new NoAvailableResourcesException(String.format(Messages.Exception.UNABLE_TO_CREATE_NETWORK_RESERVE,
-					networkOrder.getCidr()));
-		}
 
+		Integer addressRangeIndex = this.getAddressRangeIndex(virtualNetwork, subnetInfo.getLowAddress(), size);
+		String addressRangeId = this.getAddressRangeId(virtualNetwork, addressRangeIndex, cidr);
 		String ip = this.getNextAvailableAddress(virtualNetwork, addressRangeIndex);
```

Comment:
My suggestion is that these lines (from 84 to 92) be migrated to the _getCreateNetworkReserveRequest_ method which would only contain the _virtualNetwork_ parameter.

Reply:
done